blockchain/blockchain_test.go

```diff
@@ -724,3 +724,55 @@ func TestSubscribeL1Head(t *testing.T) {
 	require.True(t, ok)
 	assert.Equal(t, l1Head, got)
 }
+
+func fetchStateUpdatesAndBlocks(samples int) ([]*core.StateUpdate, []*core.Block, error) {
+	client := feeder.NewClient(utils.Mainnet.FeederURL)
+	gw := adaptfeeder.New(client)
+
+	suList := make([]*core.StateUpdate, samples)
+	blocks := make([]*core.Block, samples)
+	for i := range samples {
```

Comment:
Setup time can take time. If it's the case, we can consider using `pool` or `iter` to query concurrently with rate limit.